src/ray/gcs/redis_context.cc

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "ray/gcs/redis_context.h"
+
+#include <unistd.h>
+
+extern "C" {
+#include "hiredis/async.h"
+#include "hiredis/hiredis.h"
+#include "hiredis/adapters/ae.h"
+}
+
+namespace ray {
+
+namespace gcs {
+
+void GlobalRedisCallback(void* c, void* r, void* privdata) {
```

Comment:
how about `private_data` instead of `privdata`?

Reply:
This is the canonical way these parameters are names in hiredis, so we should keep it that way.